### PR TITLE
ecs: Add internal doc and cfn template for ECS Prometheus

### PR DESCRIPTION
**Pull request by pingleig**
**Description:** 

Only apply to Container Insights ECS Prometheus

- Add cfn template
- Add doc for some tricky parts and future work

btw: user doc will be another PR https://github.com/aws-otel/aws-otel.github.io/pull/128 ~~https://github.com/aws-otel/aws-otel.github.io draft is in https://github.com/pingleig/aws-otel.github.io/tree/ecssd-draft/src/docs/getting-started/container-insights~~

**Link to tracking Issue:** <Issue number if applicable>

- #412 

**Testing:**

N/A

**Documentation:**

This is a doc PR

### CODE REVIEW

File: deployment-template/ecs/aws-otel-container-insights-prometheus-ec2-deployment-cfn.yaml [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    Default: Default
+    Description: Enter the role arn you want to use as the ecs task role
+  ExecutionRoleArn:
+    Type: String
+    Default: Default
+    Description: Enter the role arn you want to use as the ecs execution role
+  # Collector
+  CollectorImage:
+    Type: String
+    Default: 'amazon/aws-otel-collector:latest'
```
Comment (mxiamxia) [inline deployment-template/ecs/aws-otel-container-insights-prometheus-ec2-deployment-cfn.yaml:25]: could you update to ECR repo?

File: docs/developers/container-insights-ecs-prometheus.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
+# Container Insight ECS Prometheus
```
Comment (mxiamxia) [inline docs/developers/container-insights-ecs-prometheus.md:1]: can add a link to this user guide in main README page?

Comment (pingleig) [reply docs/developers/container-insights-ecs-prometheus.md:1]: I think we don't have a section for dev doc in README, I could add new section though.

File: deployment-template/ecs/aws-otel-container-insights-prometheus-ec2-deployment-cfn.yaml (hunk 1 above)
Comment (pingleig) [reply deployment-template/ecs/aws-otel-container-insights-prometheus-ec2-deployment-cfn.yaml]: done

File: docs/developers/container-insights-ecs-prometheus.md (hunk 2 above)
Comment (pingleig) [reply docs/developers/container-insights-ecs-prometheus.md:1]: See latest README

**Review by mxiamxia — APPROVED** [review]
thanks

**Pull request merged**